spec/classes/foreman_proxy__proxydhcp__spec.rb

```diff
@@ -194,6 +194,60 @@
             'interfaces' => ['eth0']
         ) end
       end
+
+      context "as manager of ACLs for dhcp" do
+        let :facts do
+          facts.merge({:ipaddress_eth0 => '192.168.100.21',
+                       :ipaddress      => '192.168.100.21',
+                       :netmask_eth0   => '255.255.255.0',
+                       :network_eth0   => '192.168.100.0',
+                       :osfamily       => 'RedHat'})
+        end
+
+        let :pre_condition do
+          "class {'foreman_proxy':
+            dhcp_range        => '192.168.100.1 192.168.100.10',
+            dhcp_node_type    => 'secondary',
+            dhcp_peer_address => '192.168.1.20',
+            dhcp_manage_acls  => true,
+          }"
+        end
+
+        it do should contain_exec('setfacl_etc_dhcp').
+          with_command("setfacl -R -m u:foreman-proxy:rx /etc/dhcp")
+        end
+
+        it do should contain_exec('setfacl_var_lib_dhcp').
+          with_command("setfacl -R -m u:foreman-proxy:rx /var/lib/dhcpd")
+        end
+      end
+
+      context "does not manage dhcp ACLs on non-RedHat system" do
+        let :facts do
+          facts.merge({:ipaddress_eth0 => '192.168.100.21',
+                       :ipaddress      => '192.168.100.21',
+                       :netmask_eth0   => '255.255.255.0',
+                       :network_eth0   => '192.168.100.0',
+                       :osfamily       => 'Debian'})
+        end
+
+        let :pre_condition do
+          "class {'foreman_proxy':
+            dhcp_range        => '192.168.100.1 192.168.100.10',
+            dhcp_node_type    => 'secondary',
+            dhcp_peer_address => '192.168.1.20',
+            dhcp_manage_acls  => true,
+          }"
+        end
+
+        it do should_not contain_exec('setfacl_etc_dhcp').
+          with_command("setfacl -R -m u:foreman-proxy:rx /etc/dhcp")
```

Comment:
On a ```should_not``` test you can leave off the command, otherwise the exec might be present but with a different command.